nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/cmd/exec/internal/search/DefaultSearchFunctionHandlerLookup.java

```diff
@@ -95,22 +99,28 @@ private R getStaticParamValues(List<StaticCodeValue> searchResult, Command cmd)
 	
 	private R getDynamicParamValues(LookupSearchCriteria lookupSearchCriteria, Class<?> criteriaClass, List<?> searchResult) {
 		List<String> list = new ArrayList<String>(lookupSearchCriteria.getProjectCriteria().getMapsTo().values());
-		
+
 		if(list.size() > 2)
-			throw new IllegalStateException("ParamValues lookup failed due to more than 2 fields provided to create the param values. the criteria class is "+criteriaClass);
-		
-		PropertyDescriptor codePd = BeanUtils.getPropertyDescriptor(criteriaClass, list.get(0));
-		PropertyDescriptor labelPd = BeanUtils.getPropertyDescriptor(criteriaClass, list.get(1));
+		throw new IllegalStateException("ParamValues lookup failed due to more than 2 fields provided to create the param values. the criteria class is "+criteriaClass);
 		
+		String cd = list.get(0).replaceAll(toReplace, replaceWith);
+		String lb = list.get(1).replaceAll(toReplace, replaceWith);
 		try {
 			List<ParamValue> paramValues = new ArrayList<>();
 			for(Object model: searchResult) {
-				paramValues.add(new ParamValue(codePd.getReadMethod().invoke(model).toString(), (String)labelPd.getReadMethod().invoke(model)));
+				StandardEvaluationContext context = new StandardEvaluationContext(model);
```

Comment:
instead of instantiating the SpEL context here, we could use the **ExpressionEvaluator** interface which has the implementation for SpEL (**SpelExpressionEvaluator**). Inject this dependency in **DefaultSearchFunctionHandlerLookup** and invoke the getValue methods passing required parameters.